src/core/Akka/Actor/Stash/StashFactory.cs

```diff
@@ -48,7 +48,7 @@ public static class StashFactory
         /// <returns>TBD</returns>
         public static IStash CreateStash(this IActorContext context, Type actorType)
         {
-            if (actorType.Implements<IWithBoundedStash>())
+            if (actorType.Implements<IWithStash>())
                 return new BoundedStashImpl(context);
```

Comment:
`IWithStash` must resolve to an unbounded deque mailbox by default, which is how it is configured in the hocon ([L466](https://github.com/akkadotnet/akka.net/blob/76baf9d19c5c93e5377ad1ba19c0f2fedc47457f/src/core/Akka/Configuration/akka.conf#L446))

@Aaronontheweb I don't think this code has ever worked as intended, because `BoundedStashImpl` and `UnboundedStashImpl` are the exact same implementation.

Unless I'm missing something we could simply it:

```
public static class StashFactory
{
    private class StashSupport : AbstractStash
    {
        public StashSupport(IActorContext context)
            : base(context)
        { }
    }

    public static IStash CreateStash(this IActorContext context) => new StashSupport(context);
}
```

Reply:
`IWithStash` resolves to an unbounded deque by default, which is why is falling. It is not a replacement for `IWithBoundedStash`. See its [documentation](https://getakka.net/api/Akka.Actor.IWithStash.html).